workflow-templates/deploy-preview-pkgdown.yml

```diff
@@ -0,0 +1,91 @@
+on:
+  pull_request:
+    branches: [$default-branch]
+  issue_comment:
+    types: [created]
+
+name: Pkgdown Deploy Preview
+
+jobs:
+  delete-bot-comments:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Delete bot comments
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { data: comments } = await github.rest.issues.listComments({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+            });
+
+            const botUsername = 'github-actions[bot]';
+            for (const comment of comments) {
+              if (comment.user.login === botUsername) {
+                console.log(`Deleting comment ${comment.id}`);
+                await github.rest.issues.deleteComment({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  comment_id: comment.id
+                });
+              }
+            }
+
+  build-deploy-preview:
+    # Deploy a preview only if
+    # - the event is a PR
+    # - the PR is not from a fork
+    if: |
+      (github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork != true)
+    needs: delete-bot-comments
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Setup Pandoc
+        uses: r-lib/actions/setup-pandoc@v2
+
+      - name: Setup R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::pkgdown, local::.
+          needs: website
+
+      - name: Build site
+        run: pkgdown::build_site_github_pages(new_process = FALSE, install = FALSE)
+        shell: Rscript {0}
+
+      - name: Deploy Preview to Netlify as preview
+        id: netlify-deploy
+        uses: nwtgck/actions-netlify@v2
+        env:
+          NETLIFY_SITE_ID: <your netlify id>
```

Comment:
Do we need to fill this in?

Reply:
no - it's there for the person using the template to fill in with the site id for their netlify site